ClinGen/clincoded
# 2133 - add non-published sources to Case/Segregation

The changes are done according to https://docs.google.com/presentation/d/1YrVPmZVlZ5OdlAZpUKw83m35yrhS86cbdOGO1L4C-wc/edit#slide=id.g83d07f8f78_0_34

Steps to test:
1. Create a new variant curation and an interpretation
2. Navigate to Case/Segregation tab and see "Add Evidence" buttons are disabled and a dropdown list of source is now available with "Select Source" as the default selection
3. Select any source to add new evidence
4. Check that each non-published source has the required fields and field with * is required field
     Clinical Lab - Laboratory Name*, ClinVar/GTR LabID, ClinVar Submission Accession (SCV)
     Clinic - Institutional Affiliation*, Department Affiliation, ClinVar/GTR LabID, ClinVar Submission Accession (SCV)
     Research Lab - Institutional Affiliation*, Department Affiliation/Principal Investigator, ClinVar/GTR LabID, ClinVar Submission Accession (SCV)
     Public Database - Name of Database*, Database URL, Database Variant ID, ClinVar/GTR LabID, ClinVar Submission Accession (SCV)
     Other - Describe Source*
5.  When adding a new non-published source, it will not match up with an existing source even the required field has the same value as an existing one.  Each of the source is unique.
6. Check PHI note is added to non-published source add/edit modal
7. Additional note (same note in dropdown list for "Database") is added for "Public Database" modal
8. After added new evidence and if additional data (other than required one) is entered, hover over "i" icon on the evidence name in the evidence list.  Check to see the additional data is displayed.
9.  Check the added evidence(s) are displayed in tally table on top of the page grouped by source type.
